### PR TITLE
Sub: update default parameter configurations for bluerov2

### PR DESCRIPTION
**Pull request by jaxxzer**

### CODE REVIEW

**Review by Williangalvani — APPROVED** [review]

File: Tools/Frame_params/Sub/bluerov2-3_5.params [hunk 1]
```diff
@@ -82,6 +78,7 @@ BRD_SAFETY_MASK,0
 BRD_SBUS_OUT,0
 BRD_SER1_RTSCTS,2
 BRD_SER2_RTSCTS,2
+BRD_SERIAL_NUM,0
```
Comment (magicrub) [inline Tools/Frame_params/Sub/bluerov2-3_5.params:81]: This is already a default value

File: Tools/Frame_params/Sub/bluerov2-3_5.params [hunk 2]
```diff
@@ -218,6 +215,11 @@ FS_TEMP_ENABLE,0
 FS_TEMP_MAX,62
 FS_TERRAIN_ENAB,0
 GCS_PID_MASK,0
+GND_ALT_OFFSET,0.000000000000000000
```
Comment (magicrub) [inline Tools/Frame_params/Sub/bluerov2-3_5.params:218]: This is already the default

**Review by magicrub — COMMENTED** [review]
Why are you adding obviously default values into the default param file?

**Comment by rmackay9** [comment]
Yes I agree with TomP that in general we shouldn't be including default values in parameter files.  There are two good reasons:

1. when the defaults change it is unlikely that someone will remember to go back and modify these files.  This will lead to incorrect defaults being saved to users' eeproms
2. float values may be slightly different than the defaults meaning they will consume a spot in eeprom when they shouldn't.  A bit like an overbooked airplane, existing autopilot eeproms are not actually large enough to store a value for every parameter we have so it is possible to end up with a board that can no longer save new parameter values

**Comment by peterbarker** [comment]
@rmackay9 A while back I created something in the autotest suite to detect parameters which were set to the default values for the frame.  Did pick up some stuff which I PR'd at the time, but the branch didn't go anywhere.  https://github.com/peterbarker/ardupilot/tree/wip/check-frame-params With improvements that have since been made to autotest it might be feasible to push ahead with it.

**Comment by rmackay9** [comment]
@peterbarker, OK, that seems like a decent idea to me but before we do that it would be good if we had agreement from @jaxxzer.  In the end he's the owner of these param files.

**Comment by jaxxzer** [comment]
The default parameter values are intentionally included. These files are loaded to put the vehicle and all parameters into a known/expected state for the BlueROV2. Perhaps our use case is different from your experience, but this is how we have provided users a 'reset to factory default' option for a long time now.

**Pull request merged**